nodejs/eks/cluster.ts

```diff
@@ -219,7 +219,10 @@ export function generateKubeconfig(
     }
 
     if (includeProfile && opts?.profileName) {
-        env.push({ name: "AWS_PROFILE", value: opts.profileName });
+        // Use --profile instead of AWS_PROFILE because the latter can be
+        // overridden by ambient credentials:
+        // https://docs.aws.amazon.com/cli/latest/topic/config-vars.html#id1
+        args = [...args, "--profile", opts.profileName];
```

Comment:
The only potentially user-facing change.

Reply:
This change sounds good to me!

Tbh, I'm a bit surprised that even `aws eks update-kubeconfig` uses the env variable for specifying the profile.

I think it would make sense to extend the `aws-profile` test to actually deploy something to the cluster. That way we also cover the change with upgrade tests to ensure this doesn't cause any cascading replacements

Reply:
Yeah, based on https://github.com/aws/aws-cli/issues/7794 and https://github.com/aws/aws-cli/issues/4337it I'm even more convinced `--profile` is the correct behavior here.

I'll look into the upgrade test to make sure this doesn't trigger replacements. I don't think it will since the k8s provider primarily looks at the cluster/context (not user, which this impacts), but if it does we should be able to leverage the new `clusterIdentifier` config to pin the provider to the cluster.

Reply:
> I think it would make sense to extend the aws-profile test to actually deploy something to the cluster. That way we also cover the change with upgrade tests to ensure this doesn't cause any cascading replacements

@flostadler I re-enabled this upgrade test and recorded a baseline from master with a ConfigMap deployed into the cluster. It passes locally for me and I confirmed manually that no replacements were triggered.